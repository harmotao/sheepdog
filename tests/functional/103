#!/bin/bash

# Test auto recovery for store_driver(tree)

. ./common

for i in 0 1 2 3 ; do
    _start_sheep $i
done
_wait_for_sheep 4
_cluster_format -c 3 -b tree

$DOG vdi create test 40M

for i in `seq 0 4`; do
    echo $i | $DOG vdi write test $((i * 4 * 1024 * 1024)) 512
done

$DOG vdi list | _filter_short_date
$DOG node list
_node_info

echo "check remove node"
_kill_sheep 3
_wait_for_sheep_stop 3
sleep 5

echo "node=3 down"
$DOG node list
_node_info

echo "write data while node stopped"
for i in `seq 5 9`; do
    echo $i | $DOG vdi write test $((i * 4 * 1024 * 1024)) 512
done
echo "create vdi while node stopped"
$DOG vdi snapshot test
$DOG vdi clone -s 1 test test2
$DOG vdi create test3 40M -P

$DOG vdi list | _filter_short_date
$DOG node list
_node_info

echo "check recovery node"
_start_sheep 3
_wait_for_sheep 4

echo "node=3 up"
$DOG node list
_node_info

