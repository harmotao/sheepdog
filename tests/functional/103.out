QA output created by 103
using backend tree store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   40 MB   20 MB  0.0 MB DATE   7c2b25      3                22
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
   3   127.0.0.1:7003      	128          3
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	4	0	0	0	0	0
1	4	1	0	0	0	0
2	3	1	0	0	0	0
3	4	1	0	0	0	0
check remove node
node=3 down
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	5	1	0	0	0	0
1	5	1	0	0	0	0
2	5	1	0	0	0	0
3	4	1	0	0	0	0
write data while node stopped
create vdi while node stopped
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
s test         1   40 MB   40 MB  0.0 MB DATE   7c2b25      3                22
  test         0   40 MB  0.0 MB   40 MB DATE   7c2b26      3                22
  test3        0   40 MB   40 MB  0.0 MB DATE   fd3662      3                22
c test2        0   40 MB  0.0 MB   40 MB DATE   fd3815      3                22
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	20	4	0	0	0	0
1	20	4	0	0	0	0
2	20	4	0	0	0	0
3	4	1	0	0	0	0
check recovery node
node=3 up
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
   3   127.0.0.1:7003      	128          3
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	14	3	0	0	0	0
1	17	4	0	0	0	0
2	16	3	0	0	0	0
3	13	2	0	0	0	0
